### PR TITLE
network: port to extract_first_word

### PR DESCRIPTION
**Pull request by ssahani**
Port  to extract_first_word from FOREACH_WORD_QUOTED

### CODE REVIEW

**Comment by teg** [comment]
I think this needs a rebase after Lennart's recent splits, no?

**Comment by ssahani** [comment]
Done .

**Comment by poettering** [comment]
LGTM, but I'll leave this one to @teg to merge if he likes it too.

File: src/network/networkctl.c [hunk 1]
```diff
@@ -956,14 +954,19 @@ static int link_lldp_status(int argc, char *argv[], void *userdata) {
...
                 STRV_FOREACH(s, l) {
-                        FOREACH_WORD_QUOTED(word, ll, *s, state) {
-                                _cleanup_free_ char *t = NULL, *a = NULL, *b = NULL;
 
-                                t = strndup(word, ll);
-                                if (!t)
-                                        return -ENOMEM;
+                        p = *s;
+                        for (;;) {
+                                _cleanup_free_ char *t = NULL, *a = NULL, *b = NULL, *word = NULL;
```
Comment (davidreynolds) [inline src/network/networkctl.c]: Is `t` needed anymore?

**Pull request merged**